02_erc20/contract.etch

```diff
@@ -84,27 +143,57 @@ function balanceOf(owner: Address) : UInt256
     return UInt256(0u64);
   endif
 
-  return balance_state.get();
+  return balance_state.get(UInt256(0u64));
 endfunction
 
 // transfer(address _to, uint256 _value) public returns (bool success) [Send _value amount of tokens to address _to]
+// Note that the Fetch ledger does not have an implicit sender address. Hence, we need a different function signature
+@action
+function transfer(sender: Address, to: Address, value: UInt256) : Bool
+  if(!sender.signedTx())
+    return false;
+  endif
+
+  var from = sender;
+  return _transfer(from, to, value);
+endfunction
+
 // transferFrom(address _from, address _to, uint256 _value) public returns (bool success)[Send _value amount of tokens from address _from to address _to]
-// Not that the Fetch ledger does not have an implicit from address. Hence, we need a 
-// different function signature
+// Note that the Fetch ledger does not have an implicit sender address. Hence, we need a different function signature
 @action
-function transfer(from: Address, to: Address, value: UInt256) : Bool
-  if(!from.signedTx())
+function transferFrom(sender: Address, from: Address, to: Address, value: UInt256) : Bool
```

Comment:
Note, how using `sender` then allows a straightforward porting over of `transferFrom`, meaning we maintain the same interface (apart from the necessarily introduced `sender`).